src/components/Summary.js

```diff
@@ -0,0 +1,110 @@
+import React, { useState, useEffect, useRef } from "react";
+import DatePicker from "./DatePicker";
+import moment from "moment";
+
+const filterNonWorkdays = (reportedData, holidayDays) => {
+  return reportedData.filter((r) => {
+    const eventDate = new Date(r.event_date);
+    if (eventDate.getUTCDay() >= 6) {
+      // Is weekend
+      return false;
+    }
+
+    if (holidayDays.includes(r.event_date)) {
+      // Is public holiday
+      return false;
+    }
+
+    return true;
+  });
+};
+
+const loadData = async (backendUrl, date, setUsers) => {
+  if (!date) return;
+  const startDate = `${date}-01`;
+  const endDate = `${date}-31`;
+  const [names, locks, workdays] = await Promise.all([
+    fetch(`${backendUrl}/users`).then((r) => r.json()),
+    fetch(`${backendUrl}/locks/dates/${date}`).then((r) => r.json()),
+    fetch(`https://api2.codelabs.se/${date}.json`).then((r) => r.json()),
+  ]);
+
+  const holidayDays = workdays.helgdagar.map((h) => h.datum);
+
+  for (let id in names) {
+    setTimeout(async () => {
+      const reportedData = await fetch(
+        `${backendUrl}/users/${id}/events?from=${encodeURIComponent(
+          startDate
+        )}&to=${encodeURIComponent(endDate)}`
+      ).then((r) => r.json());
+
+      const reports = filterNonWorkdays(reportedData, holidayDays);
+      const isLocked = locks.some((l) => l.user_id === id);
+
+      setUsers((prevState) => {
+        return prevState.concat([
+          { name: names[id], id: id, isLocked, reports },
+        ]);
+      });
+    }, 0);
+  }
+};
+
+export function Summary({ backendUrl }) {
+  const [users, setUsers] = useState([]);
+  const [date, setDate] = useState(moment().format("YYYY-MM"));
+  const [copied, setCopied] = useState(false);
+
+  useEffect(() => {
+    setUsers([]);
+    loadData(backendUrl, date, setUsers);
+  }, [date]);
+
+  const summaryRef = useRef();
+
+  const handleDateChange = (p) => {
+    const selectedMonth = moment.unix(p / 1000).format("YYYY-MM");
+    setDate(selectedMonth);
+  };
+
+  const onCopy = (e) => {
+    navigator.clipboard
+      .writeText(summaryRef.current.innerText)
+      .then(() => setCopied(() => true))
+      .catch(() => alert("Failed to copy!"));
+  };
+
+  return (
+    <div className="summary-container">
+      <DatePicker handleDateChange={handleDateChange} />
+
+      <button className="summary-copy" onClick={onCopy}>
+        Copy
+      </button>
+      {copied ? <span style={{ color: "red" }}>Copied.</span> : null}
```

Comment:
This never gets cleared. I.e it will display the same "copied" message when you select a new month.

Reply:
Added a timer to hide it after 5 seconds